src/librustc_target/asm/nvptx.rs

```diff
@@ -0,0 +1,90 @@
+use super::{InlineAsmArch, InlineAsmType};
+use rustc_macros::HashStable_Generic;
+use std::fmt;
+
+def_reg_class! {
+    Nvptx NvptxInlineAsmRegClass {
+        reg16,
+        reg32,
+        reg64,
+        freg32,
+        freg64,
```

Comment:
If I understand PTX correctly, we only really need register classes for the `b16`, `b32` and `b64` register types. AFAIK PTX doesn't actually care about the type of a register beyond its size.

Reply:
done